### ISSUE TITLE
Controller services down

### ISSUE THREAD

**Issue by radouani1984** [mmumshad/kubernetes-the-hard-way#206]
Hello, I have installed the controller services as described in the document. However, I got an error when I attempted to check the status of the services.

`vagrant@master-1:~$ kubectl get componentstatuses --kubeconfig admin.kubeconfig`

Output

```
NAME                 STATUS      MESSAGE                                                                                     ERROR
scheduler            Unhealthy   Get http://127.0.0.1:10251/healthz: dial tcp 127.0.0.1:10251: connect: connection refused   
controller-manager   Unhealthy   Get http://127.0.0.1:10252/healthz: dial tcp 127.0.0.1:10252: connect: connection refused   
etcd-0               Healthy     {"health":"true"}                                                                           
etcd-1               Healthy     {"health":"true"} 
```

The same error goes for the master-2.
Any idea how to resolve this issue?
Thanks

**Comment by willie-cadete** [comment]
Try to check out the logs in /var/log/syslog and service status

    $ sudo systemctl status kube-controller-manager.service

    $ sudo systemctl status kube-scheduler

    $ sudo tail -n50 /var/log/syslog

**Comment by lenninmarte** [comment]
Was able to get the kube-controller-manager service running by adding the following flag to the service unit file:

--master=127.0.0.1:6443

(if adding to the /etc/systemd/system/kube-controller-manager.service file, and it is not the last argument, remember to add the trailing back-slash if keeping it organized by splitting it into different lines. That is:  --master=127.0.0.1:6443 \

Leaving it here in case someone is facing the error "invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable.
Otherwise, check the logs by using the command: systemctl -u kube-controller-manager
Then pressing [Shift + G] to go to the last line.

**Comment by kilyes** [comment]
this link helped me 
https://github.com/mmumshad/kubernetes-the-hard-way/issues/66

**Comment by fireflycons** [comment]
Hi @radouani1984 @kilyes @willie-cadete 

We have now merged a major change to bring this to v1.24.
Many issues have been addressed as part of the upgrade. Please try it now and feel free to raise further issues.

Thanks.
